src/roman_datamodels/datamodels/_datamodels.py

```diff
@@ -0,0 +1,190 @@
+        """Register each subclass in the __all__ for this module"""
+        super().__init_subclass__(**kwargs)
+        if cls.__name__ in __all__:
+            raise ValueError(f"Duplicate model type {cls.__name__}")
+
+        __all__.append(cls.__name__)
+
+
+class MosaicModel(_DataModel):
+    _node_type = stnode.WfiMosaic
+
+
+class ImageModel(_DataModel):
+    _node_type = stnode.WfiImage
+
+
+class ScienceRawModel(_DataModel):
+    _node_type = stnode.WfiScienceRaw
+
+
+class RampModel(_DataModel):
+    _node_type = stnode.Ramp
+
+    @classmethod
+    def from_science_raw(cls, model):
+        """
+        Construct a RampModel from a ScienceRawModel
+
+        Parameters
+        ----------
+        model : ScienceRawModel or RampModel
+            The input science raw model (a RampModel will also work)
+        """
+
+        if isinstance(model, cls):
+            return model
+
+        if isinstance(model, ScienceRawModel):
+            from roman_datamodels.maker_utils import mk_ramp
+
+            instance = mk_ramp(shape=model.shape)
+
+            # Copy input_model contents into RampModel
+            for key in model:
+                # If a dictionary (like meta), overwrite entries (but keep
+                # required dummy entries that may not be in input_model)
+                if isinstance(instance[key], dict):
+                    instance[key].update(getattr(model, key))
+                elif isinstance(instance[key], np.ndarray):
+                    # Cast input ndarray as RampModel dtype
+                    instance[key] = getattr(model, key).astype(instance[key].dtype)
+                else:
+                    instance[key] = getattr(model, key)
+
+            return cls(instance)
+
+        raise ValueError("Input model must be a ScienceRawModel or RampModel")
+
+
+class RampFitOutputModel(_DataModel):
+    _node_type = stnode.RampFitOutput
+
+
+class AssociationsModel(_DataModel):
+    # Need an init to allow instantiation from a JSON file
+    _node_type = stnode.Associations
+
+    @classmethod
+    def is_association(cls, asn_data):
+        """
+        Test if an object is an association by checking for required fields
+
+        Parameters
+        ----------
+        asn_data :
+            The data to be tested.
+        """
+        return isinstance(asn_data, dict) and "asn_id" in asn_data and "asn_pool" in asn_data
+
+
+class GuidewindowModel(_DataModel):
+    _node_type = stnode.Guidewindow
+
+
+class FlatRefModel(_DataModel):
+    _node_type = stnode.FlatRef
+
+
+class DarkRefModel(_DataModel):
+    _node_type = stnode.DarkRef
+
+
+class DistortionRefModel(_DataModel):
+    _node_type = stnode.DistortionRef
+
+
+class GainRefModel(_DataModel):
+    _node_type = stnode.GainRef
+
+
+class IpcRefModel(_DataModel):
+    _node_type = stnode.IpcRef
+
+
+class LinearityRefModel(_DataModel):
+    _node_type = stnode.LinearityRef
+
+    def get_primary_array_name(self):
```

Comment:
This seems a little misleading, since this is an override rather than implying that it may be overridedn. Same for the following cases  where this method is defined.

Reply:
I didn't add those docstrings that's what was there before I made my changes.